src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -281,68 +281,63 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
+        : SendFormFields.CryptoAmount
+    if (inputValue === '') {
+      // Don't show an error message when the input is empty
+      setValue(SendFormFields.AmountFieldError, '')
+      setLoading(false)
+      // Set value of the other input to an empty string as well
+      setValue(key, '')
+      return
+    }
+    const amount =
+      fieldName === SendFormFields.FiatAmount
+        ? bnOrZero(bn(inputValue).div(price)).toString()
+        : bnOrZero(bn(inputValue).times(price)).toString()
 
-        setValue(key, amount)
+    setValue(key, amount)
 
-        let estimatedFees
+    let estimatedFees
 
-        try {
-          estimatedFees = await estimateFormFees()
-          setValue(SendFormFields.EstimatedFees, estimatedFees)
-        } catch (e) {
-          setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-          setLoading(false)
+    try {
+      estimatedFees = await estimateFormFees()
+      setValue(SendFormFields.EstimatedFees, estimatedFees)
+    } catch (e) {
+      setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
+      setLoading(false)
 
-          throw e
-        }
+      throw e
+    }
 
-        const values = getValues()
+    const values = getValues()
 
-        const hasValidBalance = cryptoHumanBalance.gte(values.cryptoAmount)
-        const hasEnoughNativeTokenForGas = nativeAssetBalance
-          .minus(estimatedFees.fast.txFee)
-          .isPositive()
+    const hasValidBalance = cryptoHumanBalance.gte(values.cryptoAmount)
+    const hasEnoughNativeTokenForGas = nativeAssetBalance
+      .minus(estimatedFees.fast.txFee)
+      .isPositive()
+
+    if (!hasValidBalance) {
+      setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
+    } else if (!hasEnoughNativeTokenForGas) {
+      setValue(SendFormFields.AmountFieldError, [
+        'modals.send.errors.notEnoughNativeToken',
+        { asset: feeAsset.symbol }
+      ])
+    } else {
+      // Remove existing error messages because the send amount is valid
+      setValue(SendFormFields.AmountFieldError, '')
+    }
+    setLoading(false)
+  }
 
-        if (!hasValidBalance) {
-          setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-        } else if (!hasEnoughNativeTokenForGas) {
-          setValue(SendFormFields.AmountFieldError, [
-            'modals.send.errors.notEnoughNativeToken',
-            { asset: feeAsset.symbol }
-          ])
-        } else {
-          // Remove existing error messages because the send amount is valid
-          setValue(SendFormFields.AmountFieldError, '')
-        }
-        setLoading(false)
-      },
-      1000,
-      { leading: true }
-    ),
-    [asset, fieldName, setValue, estimateFormFees, getValues, cryptoHumanBalance, fiatBalance]
-  )
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleInputChange = useMemo(() => debounce(inputHandler, 1000, { leading: true }), [])
```

Comment:
Could we keep the `useCallback` for that since it's a semantically better fit? `useCallback(callback, [...dependencies])` is effectively equal to `useMemo(() => callback, [...dependencies])`
From what I understand, what solved it is abstracting away the `inputHandler` and memoizing the `debounce()` call itself, so it should make no difference whether to use `useMemo()` vs `useEffect()`

Reply:
`useEffect` on lodash's `debounce` doesn't work, but we can (and probably should) add a `useCallback` on the `inputHandler` - added in https://github.com/shapeshift/web/pull/1399/commits/243f20186ad5f654461fe1b146bf13ef64e04d5e.

Reply:
Actually, it's probably because there was a `useCallback` inside a `useCallback`. Anyway, I believe my latest commit addresses your thoughts!